odl/solvers/functional/functional.py

```diff
@@ -822,6 +823,83 @@ def __str__(self):
                                           self.translation)
 
 
+class InfimalConvolution(Functional):
+
+    """The functional representing ``h(x) = inf_y f(x-y) + g(y)``."""
+
+    def __init__(self, funcA, funcB):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        funcA : `Functional`
+            Function corresponding to ``f``.
+        funcB : `Functional`
+            Function corresponding to ``g``.
+
+        Example
+        -------
+        >>> space = odl.rn(3)
+        >>> l1 = odl.solvers.L1Norm(space)
+        >>> l2 = odl.solvers.L2Norm(space)
+        >>> f = odl.solvers.InfimalConvolution(l1, l2)
+        >>> f.convex_conj(0.5*f.domain.one())
+        >>> 0
+        """
+        if not isinstance(funcA, Functional):
+            raise TypeError('`func` {} is not a `Functional` instance'
+                            ''.format(funcA))
+
+        if not isinstance(funcB, Functional):
+            raise TypeError('`func` {} is not a `Functional` instance'
+                            ''.format(funcB))
+
+        self.__functional_left = funcA
+        self.__functional_right = funcB
+
+        super().__init__(space=funcA.domain,
+                         linear=False,
+                         grad_lipschitz=np.nan)
+
+    @property
+    def functional_left(self):
```

Comment:
We call these `left`/`right` in other places. See

https://github.com/odlgroup/odl/blob/20f0ce38909f4b8ae77761ceefa353be2c8b1d6c/odl/operator/operator.py#L1146-L1154